app/parse.py

```diff
@@ -1,12 +1,53 @@
 from dataclasses import dataclass
 
+import requests
+from bs4 import Tag, BeautifulSoup
+
+
+MATE_URL = "https://mate.academy/"
+
 
 @dataclass
 class Course:
     name: str
     short_description: str
     duration: str
+    modules: int = None
+    topics: int = None
+
+
+def get_detail_course_info(course_detail_url: str) -> tuple[int, int]:
+    detail_page = requests.get(MATE_URL + course_detail_url[1:]).content
+    soup = BeautifulSoup(detail_page, "html.parser")
+
+    program = soup.select_one(
+        "section.profession-page-container > "
+        "div.CourseProgram_content__tvxzY"
+    )
+    modules = len(program.select("div.CourseProgram_modules__GA_PJ > ul > li"))
```

Comment:
The code assumes that the `program` variable will always have the specified structure and that the `select` method will return a non-empty list. Consider adding error handling to manage cases where the structure might differ or the list is empty.